txtorcon/endpoints.py

```diff
@@ -298,10 +298,13 @@ def __init__(self, reactor, config, public_port,
         self.progress_listeners = []
 
         if self.hidden_service_dir is None:
-            self.hidden_service_dir = tempfile.mkdtemp(prefix='tortmp')
-            log.msg('Will delete "%s" at shutdown.' % self.hidden_service_dir)
-            delete = functools.partial(shutil.rmtree, self.hidden_service_dir)
-            reactor.addSystemEventTrigger('before', 'shutdown', delete)
+            self.add_hidden_service_dir()
+
+    def add_hidden_service_dir(self):
+        self.hidden_service_dir = tempfile.mkdtemp(prefix='tortmp')
```

Comment:
The function is also called from the listen() method, so it is better to have the method name as _add_hidden_service_dir()